src/test/java/io/confluent/connect/validate/sink/JDBCSinkLengthCheckTaskTest.java

```diff
@@ -0,0 +1,170 @@
+        server.setPort(9001); // this is the default port
+        server.setSilent(true);
+        server.setRestartOnShutdown(false);
+        server.signalCloseAllServerConnections();
+        server.setNoSystemExit(true);
+        server.start();
+
+        // Init DB
+        Class.forName("org.hsqldb.jdbc.JDBCDriver");
+        Connection dbconn = initDatabase();
+        Statement statement = dbconn.createStatement();
+
+        //Create table in DB with restrictions
+        statement.execute("SET PROPERTY \"sql.enforce_strict_size\" TRUE");
+        statement.execute("CREATE TABLE TESTINGDB (ID VARCHAR(3) NOT NULL, NAME VARCHAR(6) NOT NULL, COMMENT VARCHAR(4) NOT NULL)");
+        dbconn.commit();
+
+    }
+
+
+    @AfterClass
+    public static void tearDown() throws IOException, SQLException {
+        Connection dbconn = initDatabase();
+        dbconn.createStatement().execute("DROP TABLE TESTINGDB");
+        dbconn.createStatement().execute("TRUNCATE SCHEMA PUBLIC RESTART IDENTITY AND COMMIT NO CHECK");
+        dbconn.createStatement().execute("SHUTDOWN");
+        dbconn.close();
+    }
+
+    @Test
+    public void testStart() throws Exception{
+
+        //Get Properties
+        Map<String, String> props = new HashMap<>();
+        props.put("connection.url", "jdbc:hsqldb:hsql://localhost:9001/TESTINGDB");
+        props.put("connector.class", "io.confluent.connect.validate.JDBCSinkLengthCheckConnector");
+        props.put("LENGTH_CHECK_DEAD_LETTER_TOPIC", "DLQTopic");
+        props.put("LENGTH_CHECK_DEAD_LETTER_TOPIC_BOOTSTRAP_SERVERS", "localhost:9092");
+        props.put("connection.user", "SA");
+        props.put("connection.password", "");
+
+
+        JDBCSinkLengthCheckTask jdbctask = new JDBCSinkLengthCheckTask();
+        jdbctask.start(props);
+
+        try {
+            assertEquals(props.get("LENGTH_CHECK_DEAD_LETTER_TOPIC"), jdbctask.deadLetterTopic);
+        }catch (Exception noDLQTopic){
+            log.debug("No DLQ topic provided");
+        }
+
+        assertNotNull(jdbctask.dlqProducer);
+
+    }
+
+
+    @Test
+    public void testPut() throws Exception{
+        utils.createTopic("DLQTopic", 3, (short) 1);
+
+        JDBCSinkLengthCheckTask jdbctask2 = new JDBCSinkLengthCheckTask();
+        //Get Properties
+        Map<String, String> props = new HashMap<>();
+        props.put("connection.url", "jdbc:hsqldb:hsql://localhost:9001/TESTINGDB");
+        props.put("connector.class", "io.confluent.connect.validate.JDBCSinkLengthCheckConnector");
+        props.put("LENGTH_CHECK_DEAD_LETTER_TOPIC", "DLQTopic");
+        props.put("LENGTH_CHECK_DEAD_LETTER_TOPIC_BOOTSTRAP_SERVERS", "localhost:9092");
+        props.put("connection.user", "SA");
+        props.put("connection.password", "");
+
+        jdbctask2.start(props);
+
+
+        String topic = "TESTINGDB";
+
+        Schema keySchema = Schema.INT64_SCHEMA;
+
+        Schema valueSchema = SchemaBuilder.struct()
```

Comment:
Add a comment on why these are failed writes. Which DB constraint is violated

Reply:
Added comments on failed writes. Resolving.